json_serializable/lib/src/type_helpers/json_converter_helper.dart

```diff
@@ -202,6 +202,14 @@ _JsonConvertData? _typeConverterFrom(
     return null;
   }
 
+  if (targetType.isNullableType && matchingAnnotations.length == 2) {
```

Comment:
What if both annotations match? What if there are more than two?

I see how this would be nice for the case you're hitting – but I worry that it has too many edge cases.

Would it be better to find the set of "better" converters in the set – and see if that set of "better" converters has a length of 1?

Reply:
I think that is behaving better than currently and don't break other things.
It's exactly trying to find a "better" converter by its field type. When there are more than two converters or both matches it would throw "Found more than one matching converter for `$targetTypeCode`."

It's fixing 1339, so I'm able to have a JsonSerializable with converters and use it on a class that contains both nullable and non-nullable of the same type.

What other "algorithm" do you suggest to find better converters? What is the factor of "betterness" of the converter? Maybe you want to suggest to check also a `jsonType`. Ok, maybe it's a good idea, but I think that the level of complexity will be increased noticeably

Reply:
Maybe implement it a
```suggestion
  if (targetType.isNullableType && matchingAnnotations.length > 1) {
  matchingAnnotations
        .removeWhere((element) => !element.fieldType.isNullableType);
```
so that it is not restricted to exactly 2 matches.

Afterwards you could favor an exact type match if the result is not unique yet. Finally, instead of throwing one could the pick the first match, assuming the results are in order of declaration.

Reply:
@t-beckmann thanks for your suggestion. @kevmoo what do you think about above solution?

Reply:
any update for this PR？

Reply:
For whatever reason, the current workaround is to define converters both as annotations and as list members, however, it must be done separately for nullable and non-nullable types, like so: 

```dart
@JsonSerializable(
  converters: [
    ExampleSerializationHandlerString(),
    ExampleSerializationHandlerBool(),
    ExampleSerializationHandlerInt(),
    ExampleSerializationHandlerDouble(),
    ExampleSerializationHandlerMap(),
  ],
)
@ExampleSerializationHandlerStringNullable()
@ExampleSerializationHandlerBoolNullable()
@ExampleSerializationHandlerIntNullable()
@ExampleSerializationHandlerDoubleNullable()
@ExampleSerializationHandlerMapNullable()
```

This is with the latest json_serializable (6.8.0) and json_annotation (4.9.0) versions at the time of this writing.